### ISSUE TITLE
VAE

### ISSUE THREAD

**Issue by ayushtues** [vlgiitr/GenZoo#3]
Ayush Mangal , Pytorch , MNIST , based on Carl Doresch VAE tutorial

**Comment by aniket-agarwal1999** [comment]
Yea you seem the first for VAE implementation. So yea it's a go-ahead from our side. Be sure to once go through the README on the structure of PR that is to be submitted. :smile:

**Comment by dakshitagrawal97** [comment]
@ayushtues Any progress in this ticket?

**Comment by ayushtues** [comment]
This is my first stand-alone project ever so I kind of followed the DCGAN implementation discussion :sweat_smile:  and wanted to make things as similar as possible

**The following things are done**  

-  The model is mainly a conv layer based on Alex net (the conv_layer - relu - batch norm) some fully connected layers and then some TransposeConv layers and some fully connected layers. Although I am using Leaky ReLU instead and Xavier initialization in the conv layers. Also dropout in the encoder but not in the decoder (something I found in aniket-agarwal1999 's repo :sweat_smile: )

- I can reconstruct training images(it's reconstructing fine ) & save them in a folder generated  by the code itself 

- Applied t-sne visualization of the latent space and it has the 10 clusters needed although 1 cluster always seems to be far away from the other 9 which are sort of gaussian as expected 

- The generated output from random gaussian inputs ..they well ...kind of look like numbers :sweat_smile: 

- Did the external config file parsing thing 

-  Added progress bar

- Tried training on google colab once and it worked well although I usually train it on my CPU. Will do the final training in the end 

**Things to do:**


- TensorboardX graphs (Kinda stuck in this)

- Making the typical VAE smooth transition between numbers grid 

- Making the entire readme :sweat_smile: 

- The generate.py file to make images from the pre-trained model

- Kind of confused about the exact filter size and channel numbers of the Convolution and TransposeConv part. So will play a little bit with it. Also might try to figure out if using Leaky ReLU and Xavier Init  and batch_norm helps or makes it worse :sweat_smile: 

- Other stuff I can't remember :sweat_smile: 


Also sorry for the long comment ... it's my first time using GitHub so am not really familiar with the norms

**Comment by dakshitagrawal97** [comment]
@ayushtues Thanks for the detailed update.  No need to be sorry for writing a long comment, helps us know exactly where you are. :)

Feel free to make a PR whenever you feel the code is ready to be merged.

You might wanna keep a log of all your experiments, and note down your observations in the Readme.

**Comment by aarushgupta** [comment]
Great work @ayushtues! Feel free to shoot us a message in case you get stuck anywhere.

**Comment by ayushtues** [comment]
Merged! Closing.
